x-pack/plugins/security_solution/public/management/components/console/components/command_input/lib/entered_input.tsx

```diff
@@ -0,0 +1,295 @@
+import React from 'react';
+import type { ArgumentSelectorWrapperProps } from '../components/argument_selector_wrapper';
+import { ArgumentSelectorWrapper } from '../components/argument_selector_wrapper';
+import type { ParsedCommandInterface } from '../../../service/types';
+import type { ArgSelectorState, EnteredCommand } from '../../console_state/types';
+
+interface InputCharacter {
+  value: string;
+  renderValue: ReactNode;
+  isArgSelector: boolean;
+  argName: string;
+  argInstance: number; // zero based
+  argState: undefined | ArgSelectorState;
+}
+
+const createInputCharacter = (overrides: Partial<InputCharacter> = {}): InputCharacter => {
+  return {
+    value: '',
+    renderValue: null,
+    isArgSelector: false,
+    argName: '',
+    argInstance: 0,
+    argState: undefined,
+    ...overrides,
+  };
+};
+
+const getInputCharacters = (input: string): InputCharacter[] => {
+  return input.split('').map((char) => {
+    return createInputCharacter({
+      value: char,
+      renderValue: char,
+    });
+  });
+};
+
+const toReactJsxFragment = (prefix: string, item: InputCharacter, index: number) => {
+  return <span key={`${prefix}.${index}.${item.value ?? '$'}`}>{item.renderValue}</span>;
+};
+
+const toInputCharacterDisplayString = (
+  includeArgSelectorValues: boolean,
+  item: InputCharacter
+): string => {
+  let response = item.value;
+
+  if (includeArgSelectorValues && item.isArgSelector) {
+    response += `="${item.argState?.valueText ?? ''}"`;
+  }
+
+  return response;
+};
+
+/**
+ * Class that manages the command entered and how that is displayed to the left and right of the cursor
+ */
+export class EnteredInput {
+  private leftOfCursorContent: InputCharacter[];
+  private rightOfCursorContent: InputCharacter[];
+  private canHaveArgValueSelectors: boolean;
+  private argState: undefined | EnteredCommand['argState'];
+
+  constructor(
+    leftOfCursorText: string,
+    rightOfCursorText: string,
+    parsedInput: ParsedCommandInterface,
+    enteredCommand: undefined | EnteredCommand
+  ) {
+    this.leftOfCursorContent = getInputCharacters(leftOfCursorText);
+    this.rightOfCursorContent = getInputCharacters(rightOfCursorText);
+
+    this.canHaveArgValueSelectors = Boolean(enteredCommand?.argsWithValueSelectors);
+
+    // Determine if any argument value selector should be inserted
+    if (parsedInput.hasArgs && enteredCommand && enteredCommand.argsWithValueSelectors) {
+      this.argState = enteredCommand.argState;
+
+      const inputPieces = [
+        {
+          input: leftOfCursorText,
+          items: this.leftOfCursorContent,
+        },
+        {
+          input: rightOfCursorText,
+          items: this.rightOfCursorContent,
+        },
+      ];
+
+      for (const [argName, argDef] of Object.entries(enteredCommand.argsWithValueSelectors)) {
+        // If the argument has been used, then replace it with the Arguments Selector
+        if (parsedInput.hasArg(argName)) {
```

Comment:
When I tested it there was an interesting issue around parsing the commands, when you have additional characters after the arguments. Here's an example where I add the letter `s` after the argument `--mock`:
- if there is at least one correct argument, all arguments will be parsed:
<img width="760" alt="image" src="https://user-images.githubusercontent.com/39014407/214366738-8dac8d20-ba07-4d7f-8d5d-9febc2af5091.png">

- but, if all occurrences have additional characters, none of them:
<img width="268" alt="image" src="https://user-images.githubusercontent.com/39014407/214367052-11fe810e-573f-4203-8725-eac13ed656ab.png">


I think the issue may be in this code part. So when the parsing of the arguments (`parseInputString()` in `parsed_command_input.ts`) finds that the input has the `--mock` argument, this part will replace all occurrences.

Reply:
What I think is going on:

you are correct that it has something to do with this check here, but its more than that I think. When we go through the list of arguments that have selectors, we first check if the argument has been used and if it has, then we start to look for it on the command input text that was entered and replacing all instances of it. When we look for that argument I'm not currently checking to ensure that the argument name appears as a "whole" word - its only checking that `--mock` appear in `update --mocks --mocks --mock` (see line`105` below), so those additional `s` are being dropped from the argument name.

I need to look more into this and see how it can be fixed. I have been thinking that perhaps the "trigger" for replacing an argument that has a selector with the value selector UI might need to check to see if the user either entered a space or `=` sign - that would tells us for sure that we're into the "argument's value portion" of the input.

I'll capture this in the working issue and will likely need to follow up with a subsequent PR to address it.